system/Session/Session.php

```diff
@@ -184,14 +184,16 @@ public function __construct(\SessionHandlerInterface $driver, $config)
 
 	/**
 	 * Initialize the session container and starts up the session.
+	 *
+	 * @return mixed
 	 */
 	public function start()
 	{
 		if (is_cli() && ENVIRONMENT !== 'testing')
 		{
 			$this->logger->debug('Session: Initialization under CLI aborted.');
 
-			return;
+			return; // TODO fmertins: should we throw an exception instead return? And then @return Session phpdoc?
```

Comment:
"return" on its own is on purpose. Throwing an exception would break trying to unit test sessions.
The return returns "void", but more importantly short-circuits any additional logic. So, no it should not be tagged as returning a Session.

Reply:
Hum, ok I believe I got. My idea was avoiding return mixed, and because I saw those logs with error or warning I was thinking if it would be better generating an exception right after the log. OK, I'll  change it, tks :-)

Reply:
So, maybe a phpdoc `@return Session|void`? Because we shouldn't return anything different here... :-)

Reply:
That works

Reply:
And the todo comment should come off :-/